lib/attrsets.nix

```diff
@@ -47,6 +48,17 @@ rec {
       then hasAttrByPath (tail attrPath) e.${attr}
       else false;
 
+  /* Return if all attributes in attribute set exist.
+
+     Example:
+       x = { a = "foo"; b = "bar"; }
+       hasAttrs ["a" "b"] x
+       => true
+       hasAttrs ["a" "b" "c"] x
+       => false
+
+  */
+  hasAttrs = l: e: lib.fold (a: b: a && b) true (map (a: lib.hasAttr a e) l);
```

Comment:
There is a briefer way to write this using `lib.all`. I don't think this should be in the lib. While useful its trivial. The main issue I have with this is the name: `hasAllAttrs` seems more correct to me here.

Reply:
Fixed.

Reply:
I've removed it as it's much simpler to use the ? operator in this case.